schema/validator.go

```diff
@@ -86,13 +106,28 @@ func (v Validator) Validate(src io.Reader) error {
 	}
 }
 
-type unimplemented string
+// ValidateRefMedia validates the referenced objects against OCI media type defined in spec.
+// r: the given reader.
+// v: the expected media type.
+func ValidateRefMedia(r io.Reader, v Validator) error {
+	f, ok := vlidateRefMediaMap[v]
+	if ok {
+		if f == nil {
+			return fmt.Errorf("referenced media type validation %q unimplemented", v)
+		}
 
-func (v unimplemented) Validate(src io.Reader) error {
-	return fmt.Errorf("%s: unimplemented", v)
+		buf, err := ioutil.ReadAll(r)
+		if err != nil {
+			return errors.Wrap(err, "unable to read the document file")
+		}
```

Comment:
No need to bother with this, since you're only calling a single `f`.  Just pass your reader along:

```
if ok {
  if f == nil {
    …
  }
  return f(r)
}